### ISSUE TITLE
RCS1031: False positive on using + return in switch-statement

### ISSUE THREAD

**Issue by Debbus72** [JosefPihrt/Roslynator#632]
Based on code in our application, I have created the following example: 
```csharp
public static void Main(string[] args)
{
   switch (args[0])
   {
        default:
            { //RCS1031
                using var fs = new FileStream(@"c:\test.txt", FileMode.Open);
                return;
            }
    }
}
```
The following things happen in this example: 
- We use the new `using var ..`.
- We `return` directly from the case/default. 
Because of `using var` C# forces us to use the braces around the whole case-statement. Omitting the braces results in the error "CS8647 A using variable cannot be used directly within a switch section (consider using braces)." However, the rule "RCS1031- Remove unnecessary braces" is showing on those braces. 

We have work-around for this, like not do a `return` but a `break` (which is just below the closing-brace). The working example (which does not show the RCS1031 rule) would be:
```csharp
  default:
  {
      using var fs = new FileStream(@"c:\test2.txt", FileMode.Open);
  }
  break;
```

For me this is a real edge-case and frankly I quickly rewrote this to not do the `return` inside the switch. So see if you want/can to fix this.
